lib/block-supports/layout.php

```diff
@@ -232,10 +232,12 @@ function( $matches ) {
  * to avoid breaking styles relying on that div.
  *
  * @param string $block_content Rendered block content.
+ * @param  array  $block        Block object.
  * @return string Filtered block content.
  */
-function gutenberg_restore_image_outer_container( $block_content ) {
-	$image_with_align = '/(^\s*<figure\b[^>]*)\bwp-block-image\b([^"]*\b(?:alignleft|alignright|aligncenter)\b[^>]*>.*<\/figure>)/U';
+function gutenberg_restore_image_outer_container( $block_content, $block ) {
+	$block_classnames = isset( $block['attrs']['className'] ) ? $block['attrs']['className'] : '';
+	$image_with_align = '/(^\s*<figure\b[^>]*)\bwp-block-image\b([^"]*\b(?:alignleft|alignright|aligncenter).*)\b(?:' . $block_classnames . ')([^>]*>.*<\/figure>)/U';
```

Comment:
Is it possible that the classnames show up in a different order than the regex?

Reply:
Should the `$accepted_args` on the `add_filter` below be now `2`?

E.g.,

`add_filter( 'render_block_core/image', 'gutenberg_restore_image_outer_container', 10, 2 );`

Reply:
> Should the $accepted_args on the add_filter below be now 2

True, fixed, thanks

Reply:
> Is it possible that the classnames show up in a different order than the regex?

@youknowriad I don't know what the chances are of this, but anything is possible! I have refactored this to hopefully account for any order of the classnames, which also made it easier to get rid of those extra spaces as an added bonus 🎉